testsuite/tests/singlecluster/gateway/dnspolicy/test_dnspolicy_removal.py

```diff
@@ -0,0 +1,43 @@
+"""Test that Gateway will behave properly after attached DNSPolicy is deleted"""
+
+import pytest
+
+from testsuite.gateway.gateway_api.gateway import KuadrantGateway
+
+pytestmark = [pytest.mark.kuadrant_only, pytest.mark.dnspolicy]
+
+
+@pytest.fixture(scope="module")
+def gateway(request, cluster, blame, wildcard_domain, module_label):
+    """Create gateway without TLS enabled"""
+    gw = KuadrantGateway.create_instance(cluster, blame("gw"), wildcard_domain, {"app": module_label}, tls=False)
+    request.addfinalizer(gw.delete)
+    gw.commit()
+    gw.wait_for_ready()
+    return gw
+
+
+@pytest.fixture(scope="module", autouse=True)
+def commit(request, route, dns_policy):  # pylint: disable=unused-argument
+    """Commits dnspolicy"""
+    request.addfinalizer(dns_policy.delete)
+    dns_policy.commit()
+    dns_policy.wait_for_ready()
+
+
+def test_dnspolicy_removal(gateway, dns_policy, client):
+    """
+    Test that Gateway will behave properly after attached DNSPolicy is deleted
+    - Verify that Gateway is affected by DNSPolicy and requests are successful
+    - Delete attached DNSPolicy
+    - Verify that Gateway is no longer affected by DNSPolicy and requests are failing
+    """
+    assert gateway.refresh().is_affected_by(dns_policy)
+    response = client.get("/get")
+    assert response.status_code == 200
+
+    dns_policy.delete()
+
+    assert not gateway.refresh().is_affected_by(dns_policy)
```

Comment:
This assertion is not sufficient. I am getting an error:
```
        assert not gateway.refresh().is_affected_by(dns_policy)
        response = client.get("/get")
>       assert response.has_dns_error()
E       assert False
E        +  where False = has_dns_error()
E        +    where has_dns_error = <testsuite.httpx.Result object at 0x7f54b8b28320>.has_dns_error
```

It takes some time for DNS record in cloud provider (I was using AWS) to get removed and the test is too quick so there is no DNS error there. The test works like a charm when debugging or if there is a short sleep added. Apparently `dns_policy.delete()` does not wait for related DNSRecord CR to be deleted (and for related DNS record in cloud provider to be deleted)

Reply:
Not sure if `sleep_ttl` is stable enough, see https://github.com/Kuadrant/testsuite/pull/535#discussion_r1758510522 - but I might be completely wrong on that.

Reply:
It is stable now so I approved this one, but I think that it would be better to use the same approach as @azgabur used in his PR (see the link to PR 535 comment above).